pkg/provider/aws/plugin.go

```diff
@@ -74,23 +74,54 @@ func (a *AWS) copyConstructEdgeToDag(
 	dep graph.Edge[core.Construct],
 	dag *core.ResourceGraph) error {
 	data := knowledgebase.EdgeData{AppName: a.Config.AppName, Source: sourceResource, Destination: destinationResource}
+
 	switch construct := dep.Source.(type) {
 	case *core.ExecutionUnit:
-		switch dep.Destination.(type) {
+		if a.Config.GetExecutionUnit(construct.ID).Type == kubernetes.KubernetesType {
+			data.SourceRef = construct.AnnotationKey
+		}
```

Comment:
Is the reason this is only set for k8s because it's only used for that scenario? If we really want to make this field be hyper-specific, IMO the name should reflect that.

Reply:
right now its only set for k8s because thats all that needs it, however there could be other scenarios in which we need it in the near term, we just havent run into those.

The hope is that we dont though i just named it generic in case